README.rst

```diff
@@ -12,77 +12,105 @@ smart_open — utils for streaming large files in Python
-
+  >>> from smart_open import open
+  >>>
   >>> # stream lines from an S3 object
-  >>> for line in smart_open('s3://mybucket/mykey.txt', 'rb'):
-  ...    print(line.decode('utf8'))
+  >>> for line in open('s3://commoncrawl/robots.txt'):
+  ...    print(line, end='')
+  ...    break
+  User-Agent: *
 
+  >>> gz_path = 'smart_open/tests/test_data/crime-and-punishment.txt.gz'
+  >>>
   >>> # stream from/to compressed files, with transparent (de)compression:
-  >>> for line in smart_open('./foo.txt.gz', encoding='utf8'):
-  ...    print(line)
+  >>> for line in open(gz_path, encoding='utf-8'):
+  ...    print(line[:50])
+  ...    break
+  В начале июля, в чрезвычайно жаркое время, под веч
 
   >>> # can use context managers too:
-  >>> with smart_open('/home/radim/foo.txt.bz2', 'wb') as fout:
-  ...    fout.write(u"some content\n".encode('utf8'))
-
-  >>> with smart_open('s3://mybucket/mykey.txt', 'rb') as fin:
+  >>> bz2_path = gz_path.replace('.gz', '.bz2')
+  >>> with open(gz_path) as fin:
+  ...    with open(bz2_path, 'w') as fout:
+  ...        for line in fin:
+  ...           fout.write(line)
+
+  >>> # can use any IOBase operations, like seek
+  >>> with open('s3://commoncrawl/robots.txt', 'rb') as fin:
   ...     for line in fin:
-  ...         print(line.decode('utf8'))
+  ...         print(line.decode('utf-8'), end='')
+  ...         break
   ...     fin.seek(0)  # seek to the beginning
-  ...     b1000 = fin.read(1000)  # read 1000 bytes
-
-  >>> # stream from HDFS
-  >>> for line in smart_open('hdfs://user/hadoop/my_file.txt', encoding='utf8'):
-  ...     print(line)
+  ...     print(fin.read(4))
+  User-Agent: *
+  0
+  b'User'
 
   >>> # stream from HTTP
-  >>> for line in smart_open('http://example.com/index.html'):
-  ...     print(line)
+  >>> for line in open('http://example.com/index.html'):
+  ...     print(line, end='')
+  ...     break
+  <!doctype html>
 
-  >>> # stream from WebHDFS
-  >>> for line in smart_open('webhdfs://host:port/user/hadoop/my_file.txt'):
-  ...     print(line)
+  >>> # stream from HDFS
+  >> for line in open('hdfs://user/hadoop/my_file.txt', encoding='utf8'):
+  ..     print(line)
 
-  >>> # stream content *into* S3 (write mode):
-  >>> with smart_open('s3://mybucket/mykey.txt', 'wb') as fout:
-  ...     for line in [b'first line\n', b'second line\n', b'third line\n']:
-  ...          fout.write(line)
+  >>> # stream from WebHDFS
+  >> for line in open('webhdfs://host:port/user/hadoop/my_file.txt'):
+  ..     print(line)
+
+  >>> # stream content *into* S3 (write mode), using a custom Session for authentication:
+  >>> import boto3
+  >>> tkwa = dict(session=boto3.Session(profile_name='smart_open'))
+  >>> url = 's3://smart-open-py37-benchmark-results/test.txt'
+  >>> lines = [b'first line\n', b'second line\n', b'third line\n']
+  >>> [len(l) for l in lines]
+  [11, 12, 11]
+  >>>
+  >>> with open(url, 'wb', tkwa=tkwa) as fout:
+  ...    for line in lines:
+  ...         fout.write(line)
+  11
```

Comment:
Do we really need this output in the README? It looks more confusing than illuminating to me.

Reply:
Well, we can hide it by doing a variable assignment, e.g. `bytes_written = fout.write(line)`.

The reason that output is there is because it's what the library outputs in real life if a user entered that code in a REPL. That's the beauty of the tests: it's the real deal, not an artist's impression of what the output may look like.

Reply:
This is not what I see in REPL though. What REPL are you using? It sounds bizarre that a REPL would print (??) unassigned return values.

Reply:
This is pretty common: we [recently](https://github.com/RaRe-Technologies/smart_open/issues/271) had a user ask about this particular behavior. So, I think it's something we should document in our examples.

To answer your question directly, it depends on your REPL:

```
(smart_open) misha@cabron:~$ ipython
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.0.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import random

In [2]: for _ in range(3):
   ...:     random.randint(1, 10)

In [3]:                                                                                                                                                                                                       
Do you really want to exit ([y]/n)? y
(smart_open) misha@cabron:~$ python
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import random
>>> for _ in range(3):
...     random.randint(1, 10)
... 
3
3
1
>>> 
```

So ipython suppresses that output, but the standard python REPL does not.

Reply:
Looks almost like a REPL bug to me! (printing stuff that noone asked it to print… WTH).

Either way, I'm strongly -1 on including such output in our examples or README. Very confusing, even to me.

Reply:
OK, I've suppressed printing via variable assignment.